plugins/pulp_rpm/plugins/importers/yum/existing.py

```diff
@@ -116,14 +116,20 @@ def check_all_and_associate(wanted, conduit, config, download_deferred, catalog)
                 named tuples received as input were not found on the server.
     :rtype:     set
     """
+    rpm_drpm_srpm = (ids.TYPE_ID_RPM, ids.TYPE_ID_SRPM, ids.TYPE_ID_DRPM)
+    all_associated_units = set()
+    for unit_type in rpm_drpm_srpm:
+        units_generator = repo_controller.get_associated_unit_ids(conduit.repo.repo_id, unit_type)
+        all_associated_units.update(list(units_generator))
```

Comment:
Yeah... `update` expects iterable. Thanks.
Though there still won't be two copies of data :) but two copies of references to the same data.
I will fix it in the next push.